### DISCUSSION TITLE
Some servers cannot query the information

### DISCUSSION THREAD

**Discussion by molanp** [molanp/zhenxun_plugin_mccheck#2] [General]
建议：在本地查询完毕后进行远程api调用复查

**Comment by molanp** [comment]
such as 
```
2b2t.org:25565
```
**2b2t.xin** is OK,but its server in China
```
Version: Waterfall 1.12.2
Protocol: EXTENDED_LEGACY
Server IP: 2b2t.xin
Server port: 25565
Delay: 29ms
Motd: 2B 己所不欲，勿施于人！
Players: 86/1024
Status: SUCCESS|(Request & response parsing OK)
favicon：null
```

**Comment by molanp** [comment]
The old branch([06d0b37](https://github.com/molanp/zhenxun_chafu_Minecraft/commit/06d0b3783a51e4e659331b67097133cf723445b9)) doesn't have this problem, but often timeouts

**Comment by Enderman-Teleporting** [comment]
If a forein server is able to be connected in China, I think it can be recognized. However, what I am thinking is that,the connection of certain forein servers like 2b2t.org are not stable, and the failure of the connection of 2b2t.org might be for certain plugins. I strongly recommend you to open an issue on this for the repo minestat, cus im alse experiencing such errors.

**Comment by tzdtwsj** [comment]
I think 2b2t.org has problem,too.Because 2b2t.org's DNS not found in Google DNS Server(8.8.8.8)，2b2t.org and 2b2t.xin isn't same.I can get 2b2t.xin's ip but can't get 2b2t.org.I don't know why we can connect the 2b2t.org in Minecraft.This is strange.But the old branch can get the data.This maybe because the old branch used the mcapi.us's api,not get data for itself.The mcapi.us used some means,Make it possible to get data in 2b2t.org.

**Comment by Enderman-Teleporting** [comment]
As you told us yesterday ,**@tzdtwsj**, the server `2b2t.org` cannot be pinged.
I think this is 'cus of special plugins which block the requests

**Comment by tzdtwsj** [comment]
@Enderman-Teleporting Maybe 2b2t.org used some means.Because my friend said:"Housemaster knows the CloudFlare's owner."

**Comment by Enderman-Teleporting** [comment]
Have you commented on the issue minestat#141? you may make some additional contributions

**Comment by Enderman-Teleporting** [comment]
I think this issue should be opened as a discussion instead

**Comment by tzdtwsj** [comment]
@molanp I will

**Comment by tzdtwsj** [comment]
The about 2b2t.org's problem is solved.Because 2b2t.org isn't the server IP.The true domain is _minecraft._tcp.2b2t.org,because 2b2t.org used srv analysis,no A or CNAME.
I used the command to show the _minecraft._tcp.2b2t.org's point.
```bash
nslookup -q=srv _minecraft._tcp.2b2t.org
```
The result is:
```
Server:         8.8.8.8
Address:        8.8.8.8#53

Non-authoritative answer:
_minecraft._tcp.2b2t.org        service = 10 10 25565 connect.2b2t.org.

Authoritative answers can be found from:
```

So,the 2b2t.org actually points to connect.2b2t.org.You can't get data in 2b2t.org,this is normal,because 2b2t.org isn't the server's IP.It is connect.2b2t.org.

But the project can't get the srv data.You must use connect.2b2t.org.
